simplifier/operations/bitwise.py

```diff
@@ -175,6 +175,39 @@ def negate(self, level: Optional[int] = None, predecessors: Optional[List[Operat
         self.world.replace(self, new_op)
         return new_op
 
+    def get_equivalent_condition_of_negated_type(self, operation: Union[BitwiseAnd, BitwiseOr]) -> Union[BitwiseAnd, BitwiseOr]:
+        """
+        Create the new condition that is equivalent to self and replace self by it.
+
+        - if self is Or-operation and operation is a conjunction
+            self = (a1 & a2 & ... & ak) | l1 | l2 | ... | lh, where operation = (a1 & a2 & ... & ak),
+            remaining_conditions = [l1, l2, ..., lh] and each li is a condition
+            -> new_condition = (a1 | l1 | ... |lh) & (a2 | l1 | ... |lh) & ... & (ak | l1 | ... |lh)
+        -if self is And-operation and operation is a disjunction
+            self = (a1 | a2 | ... | ak) & l1 & l2 & ... & lh, where disjunction = (a1 | a2 | ... | ak),
+            remaining_conditions = [l1, l2, ..., lh] and each li is a condition
+            -> new_condition = (a1 & l1 & ... & lh) | (a2 & l1 & ... & lh) | ... | (ak & l1 & ... & lh)
+        """
+        conditions = []
+        for literal in operation.operands:
+            new_operands = [literal, *(cond.copy_tree() for cond in self.operands if hash(cond) != hash(operation))]
+            conditions.append(self.__class__(self.world).replace_operands(new_operands))
```

Comment:
I may be out of the loop - why do we use `self.__class__(self.world)` instead of `self.copy()`?

Reply:
I have no clue, I changed it to copy.